packages/babel-helpers/src/helpers/usingCtx.ts

```diff
@@ -30,18 +45,24 @@ export default function _usingCtx() {
       // core-js-pure uses Symbol.for for polyfilling well-known symbols
       if (isAwait) {
         var dispose =
-          value[Symbol.asyncDispose || Symbol.for("Symbol.asyncDispose")];
+            value[Symbol.asyncDispose || Symbol.for("Symbol.asyncDispose")],
+          kind = DisposeKind.AWAIT_ASYNC;
       }
-      if (dispose == null) {
+      if (dispose === undefined) {
         dispose = value[Symbol.dispose || Symbol.for("Symbol.dispose")];
+        kind &= DisposeFlag.ASYNC_MASK;
       }
       if (typeof dispose !== "function") {
-        throw new TypeError(`Property [Symbol.dispose] is not a function.`);
+        throw new TypeError(
```

Comment:
Maybe just throw `Invalid dispose function`.
Because `"Property [Symbol.dispose] is not a function."` it is also possible for the user to add a `[Symbol.asyncDispose]`.

Reply:
Okay, how about "Object is not disposable.", since "dispose" is a verb.